doc/api/http2.md

```diff
@@ -558,11 +558,70 @@ added: REPLACEME
 Calls [`unref()`][`net.Socket.prototype.unref`] on this `Http2Session`
 instance's underlying [`net.Socket`].
 
+### Class: ServerHttp2Session
+<!-- YAML
+added: v8.4.0
+-->
+
+#### serverhttp2session.alternateService(alt[, origin[, stream]])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `alt` {string} A description of the alternative service configuration as
+  defined by [RFC 7838][].
+* `origin` {string} A URL string specifying the origin.
+* `stream` {number} The numeric identifier of a stream (or `0`). Defaults to
+  `0`.
+
+Submits an `ALTSVC` frame (as defined by [RFC 7838][]) to the connected client.
+
+```js
+const http2 = require('http2');
+
+const server = http2.createServer();
+server.on('session', (session) => {
+  // Set altsvc for origin https://example.org:80
+  session.altsvc('h2=":8000"', 'https://example.org:80');
+});
+
+server.on('stream', (stream) => {
+  // Set altsvc for a specific stream
+  stream.session.altsvc('h2=":8000"', undefined, stream.id);
+});
+```
+
+Sending an `ALTSVC` frame with stream ID `0` (the default) indicates that
+the alternate service is associated with the given `origin`. Sending an
+`ALTSVC` frame with a specific stream ID indicates that the alternate
+service is associated with the origin of the given `Http2Stream`.
+
 ### Class: ClientHttp2Session
 <!-- YAML
 added: v8.4.0
 -->
 
+#### Event: 'altsvc'
```

Comment:
I sort of feel this name should correspond with the name of the method on the server side. I’m ambivalent about whether alternateService is a better name than altsvc though: I feel eventually most people would refer to it as ALTSVC (HTTP/2 frame) or Alt-Svc (HTTP/1.1 header), while technically the name of the header is alternateService*s*.

Reply:
@addaleax ... given that you were the one that made the original rename suggestion, what do you think here?

Reply:
@jasnell If everybody here agrees that it’s going to be referred to as `ALTSVC` anyway, yea, probably best to stick with it?

Either way, yes, +1 for being consistent :)

Reply:
I think `altsvc` is definitely going to be the most common given the way the header and frame type are defined. I'll switch the API call back :-)